analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsShouldBeLogged.Microsoft.Extensions.Logging.cs

```diff
@@ -0,0 +1,405 @@
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogDebug(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogError(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogInformation(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogTrace(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogWarning(logger, e, "Message!");
+        }
+    }
+
+    public void NoLogsInCatch()
+    {
+        try { }
+        catch (Exception e)
+        {
+        }
+    }
+
+    public void CallingMethodToLog()
+    {
+        try { }
+        catch (Exception e)
+        {
+            Log();
+        }
+    }
+
+    private void Log()
+    {
+        logger.LogCritical("Message!"); // Compliant - we do not check this
+    }
+
+    private void LogFromMultipleCatchBlocks()
+    {
+        try { }
+        catch (DivideByZeroException)
+        {
+            LoggerExtensions.LogCritical(logger, "Message!"); // Noncompliant
+        }
+        catch (AggregateException)
+        {
+            LoggerExtensions.LogCritical(logger, "Message!"); // Noncompliant
+        }
+        catch (ApplicationException e)
+        {
+            LoggerExtensions.LogCritical(logger, e, "Message!");
+        }
+    }
+
+    private void LogFromNestedCatchBlocks(Exception wrongException)
+    {
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning("Message!");                  // Noncompliant
+            logger.LogWarning(wrongException, "Message!");  // Noncompliant - wrong exception
+            try { }
+            catch (DivideByZeroException)
+            {
+                logger.LogCritical("Message!");             // Noncompliant
+            }
+            catch (AggregateException e1)
+            {
+                logger.LogCritical(e, "Message!");          // Noncompliant - wrong exception
+            }
+            catch (Exception e2)
+            {
+                logger.LogCritical(e2, "Message!");
+            }
+        }
+    }
+
+    private void ReAssignment()
+    {
+        try { }
+        catch (Exception e)
+        {
+            var other = e;
+            logger.LogWarning(other, "Message!");     // Noncompliant
+        }
+    }
+
+    private void Filtering()
+    {
+        try { }
+        catch (Exception e) when (e is InvalidCastException)
+        {
+            logger.LogWarning(e, "Message!");
+        }
+        catch (Exception e) when (e is DivideByZeroException divideByZeroException)
+        {
+            logger.LogWarning(divideByZeroException, "Message!");   // Noncompliant - FP the exception has other name
```

Comment:
This FP can be fixed if the rule doesn't raise when:
- a logging method with `Exception` parameter is called in the catch block (doesn't matter with which `Exception` variable)
- the catch block has a `when` clause
WDYT? Is it worth the effort?

Reply:
Nice suggestion!